### PR TITLE
reduce plugin rerenders using memo

### PR DESCRIPTION
**Pull request by mejackreed**
While monitoring http://127.0.0.1:4444/__tests__/integration/mirador/plugins/add.html I'm seeing a reduction of rerenders from 1319 -> 1230

### CODE REVIEW

File: __tests__/src/extend/withPlugins.test.js [hunk 1]
```diff
@@ -19,15 +19,17 @@ function createPluginHoc(pluginMap) {
...
 
-  it('should return a function (curry function call)', () => {
-    expect(withPlugins('Target')(Target)).toBeInstanceOf(Function);
+  it('should return a React.memo object (curry function call)', () => {
+    expect(withPlugins('Target')(Target)).toBeInstanceOf(Object);
+    expect(withPlugins('Target')(Target).type).toBeInstanceOf(Function);
   });
 
-  it('displayName prop of returned function is based on target name argument', () => {
+  xit('displayName prop of returned function is based on target name argument', () => {
```
Comment (mejackreed) [inline __tests__/src/extend/withPlugins.test.js]: I couldn't figure out how to get this to return properly.

**Comment by charbugs** [comment]
Can we hold on merging this? I'm currently about a refactoring that seems promising to me in terms of performance.

**Comment by mejackreed** [comment]
Of course.

File: __tests__/src/extend/withPlugins.test.js (hunk 1 above)
Comment (charbugs) [reply __tests__/src/extend/withPlugins.test.js]: There are also other tests failing in withPlugin. I already run into similar issues with that memo function. The problem is that the memo function does not return a component function but a react element object. And this object is somewhat opaque and not described in the docs.

What about putting the PluginHoc at top level and then doing a name export for testing purpose?

**Pull request closed without merging**